modules/node_groups/launch_template.tf

```diff
@@ -75,6 +75,15 @@ resource "aws_launch_template" "workers" {
 
   key_name = lookup(each.value, "key_name", null)
 
+  dynamic "metadata_options" {
+    for_each = lookup(each.value, "metadata_options", null) != null ? [lookup(each.value, "metadata_options", null)] : []
+    content {
+      http_endpoint               = lookup(metadata_options.value, "http_endpoint", null)
+      http_tokens                 = lookup(metadata_options.value, "http_tokens", null)
+      http_put_response_hop_limit = lookup(metadata_options.value, "http_put_response_hop_limit", null)
```

Comment:
Since the defaults are primitive variables, you need to specify them in the lookups or do a merge somewhere if you want to use an object

```hcl
http_endpoint               = lookup(metadata_options.value, "http_endpoint", var.workers_group_defaults["metadata_http_endpoint"])
http_tokens                 = lookup(metadata_options.value, "http_tokens", var.workers_group_defaults["metadata_http_tokens"])
http_put_response_hop_limit = lookup(metadata_options.value, "http_put_response_hop_limit", var.workers_group_defaults["metadata_http_put_response_hop_limit"])
```

or

```hcl
metadata_options = merge(
  {
     http_endpoint               = var.workers_group_defaults["metadata_http_endpoint"]
     http_tokens                 = var.workers_group_defaults["metadata_http_tokens"]
     http_put_response_hop_limit = var.workers_group_defaults["metadata_http_put_response_hop_limit"]
  },
  lookup(each.value, "metadata_options", {}),
)
```

Reply:
I use `dynamic "metadata_options"` due to https://github.com/terraform-aws-modules/terraform-aws-eks/pull/1562#discussion_r700916693, and I missed about `workers_group_defaults` are primitives. It's my bad.
But looks like here is two ways:
1) Do not use dynamic block and use defaults (as was at https://github.com/terraform-aws-modules/terraform-aws-eks/pull/1564/commits/0873af5bcf4ca7ca57d3877caadefa77654939ae). I don't see the point of using dynamic in this case, because with _defaults_ we always have all of 3 values for `metadata_options`. I think it's bad idea to merge _defaults_, only if at least one of 3 inputs are present, and use provider default behavior if not.
2) Do not use _defaults_ for `metadata_options`. Here we can use object as an input, `dynamic "metadata_options"` block,  and create it only if an input is not empty.

Reply:
As for me - I prefer 1.

Reply:
I agree, let's close this one, and merge #1485 instead.